pkg/api/v1/types.go

```diff
@@ -1706,6 +1706,20 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+// DaemonEndpoint stores information about a single Daemon endpoint.
+type DaemonEndpoint struct {
+	// Port number of the given endpoint.
+	Port int `json:port`
+}
+
+// NodeDaemonEndpoints lists ports opened by daemons running on the Node.
+type NodeDaemonEndpoints struct {
+	// Endpoint on which Kubelet is listening.
+	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty"`
+	// Endpoint opened by Kubelet on which cAdvisor proxy is available.
+	CAdvisorProxyEndpoint DaemonEndpoint `json:"cAdvisorProxyEndpoint,omitempty"`
```

Comment:
Since it's backwards compatible to add it later if we find that we need it, what about just leaving it out for now? The important change is to get the kubelet's port visible in the api (so that it can be set to something other than the default) and I don't want to block that change on determining whether or not we need to expose cadvisor as well.